n3fit/src/n3fit/layers/DY.py

```diff
@@ -43,20 +45,20 @@ def call(self, pdf_raw):
         results = []
         if self.many_masks:
             if self.splitting:
-                splitted_pdf = op.split(pdf_raw, self.splitting, axis=1)
+                splitted_pdf = op.split(pdf_raw, self.splitting, axis=2)
                 for mask, pdf, fk in zip(self.all_masks, splitted_pdf, self.fktables):
                     pdf_x_pdf = op.pdf_masked_convolution(pdf, mask)
-                    res = op.tensor_product(fk, pdf_x_pdf, axes=3)
+                    res = op.tensor_product(fk, pdf_x_pdf, axes=[(1, 2, 3), (1, 2, 3)])
```

Comment:
I agree this looks ugly, I also didn't like the axes=3 though, always found it confusing that 3 is the amount of axes rather than an index of a particular one. But yes einsum would be much nicer.
I would propose though not to do that here, but later in a separate PR on GPU optimizations, as that will touch on exactly these things. (And I strongly suspect this will cause numerical differences, though I didn't test)

Reply:
ok! 

> always found it confusing that 3 is the amount of axes rather than an index of a particular one

For me the axes=n is the most intuitive representation instead, because it is axes not axis! :P 
But ok with leaving it for later.